python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -550,7 +636,7 @@ def run(self, exposure=None, selectSources=None, templateExposure=None, template
                         selectSources = self.subtract.getSelectSources(
                             exposure,
                             sigma=scienceSigmaPost,
-                            doSmooth=not self.doPreConvolve,
+                            doSmooth=not self.config.doPreConvolve,
```

Comment:
Has this option just been broken for a long time?

Reply:
This was spotted when I did a test run with doSelectSources == True.